src/msemu.c

```diff
@@ -806,15 +812,12 @@ int main(int argc, char *argv[])
 	 * passed, then create a new dataflash in RAM which will get written
 	 * to ./dataflash.bin
 	 */
-	flashtobuf((uint8_t *)ms.dev_map[DF], dataflash_path, MEBIBYTE/2);
-	/* XXX: Check return of this func, create new dataflash image! */
-#if 0
-	if (!ret && !opt_dataflash) {
-		generate new dataflash image
-		XXX: Generate random serial number?
-		printf("Generating new dataflash image. Saving to file: %s\n",
-		  dataflash_path);
-#endif
+	ret = flashtobuf((uint8_t *)ms.dev_map[DF], dataflash_path, MEBIBYTE/2);
+	if (ret != 0 && opt_dataflash_set) {
+		printf("Dataflash image not found at '%s', aborting.\n", dataflash_path);
+		abort();
+	}
```

Comment:
Yeah, I was just implementing it based on comment above it.

I think it makes more sense to just create the dataflash regardless if it can't be found.